exporter/redis.go

```diff
@@ -684,42 +806,39 @@ func (e *Exporter) scrapeRedisHost(scrapes chan<- scrapeResult, addr string, idx
 		}
 	}
 
+	log.Debugf("e.singleKeys: %#v", e.singleKeys)
+	allKeys := append([]dbKeyPair{}, e.singleKeys...)
+
 	log.Debugf("e.keys: %#v", e.keys)
-	for _, k := range e.keys {
+	scannedKeys, err := getKeysFromPatterns(c, e.keys)
+	if err != nil {
```

Comment:
if an error happens here, is there a point to continue with `scannedKeys` ?

Reply:
Nope; changed.